app/helpers/handlers.js

```diff
@@ -383,22 +373,24 @@ const showPostPage = (currentQuestion, request, h) => {
       }
       break
     case 'project-cost-solar':
-      const calculatedGrantCalfVar = getYarValue(request, 'calculatedGrantCalf')
       // ineligible as calf housing too low
-      if (calculatedGrantCalfVar < 15000) {
+      if (getYarValue(request, 'calculatedGrantCalf') < 15000) {
         return h.view('not-eligible', NOT_ELIGIBLE)
       // calf housing only
-      } else if (calculatedGrantCalfVar >= 500000) {
+      } else if (getYarValue(request, 'calculatedGrantCalf') >= 500000) {
         setYarValue(request, 'calculatedGrant', 500000)
         return h.redirect('/upgrading-calf-housing/potential-amount-conditional')
         // solar capping
       } else if (getYarValue(request, 'calculatedGrant') > 500000) {
-        const newCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
+        const solarCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
         // store capped solar value for potential amount solar capped page
-        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'calculatedGrantSolar'))
+        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'SolarPVCost') * 0.25)
```

Comment:
why are you calculating this value when we have the original calculatedSolarGrant value?

Reply:
it is true, yet calculatedGrantSolar is the original calculated grant with no capping, aka what the user would get. we set that to calculatedGrantSolarPreCap to show the value before the capping (the original value), then set calcualtedGrantSolar to be the users solar grant post capping, so we have both values

I'm just nto sure why you are redoing the same calculation when we already have this value :P

Reply:
we are not doing any calculation here just separating variables . if calculated grant will be capped based on rules . we still have to show uncapped value which is 25% of entered value

Reply:
we already do, its called calculatedgrantsolar

You're calculating it by mutliplying the users entered value by 0.25, thus doing the SAME CALCULATION a second time

Reply:
got your point changing code